bin/histopreprocess.py

```diff
@@ -0,0 +1,191 @@
+import os
+import random
+from pathlib import Path
+import openslide
+from PIL import Image
+import numpy as np
+import cv2
+import matplotlib.pyplot as plt
+import IPython.display as display       
+import argparse
+
+
+def main():
+    # Import slide
+    slide = openslide.OpenSlide(file_path)
+
+    # Define the level
+    level = 0
+
+    image = slide.read_region((0,0), level, slide.level_dimensions[level])
+
+    # ----- Scaling -----
+    # patch size in pixel 
+    patch_size_px = 512
+    # size thumbnail
+    thumb_size = (np.array(slide.dimensions)/patch_size_px).astype(int)
+    # creating thumbnail of the image
+    image_thumb = slide.get_thumbnail(thumb_size) 
+
+    # ----- RGB Thresholding -----
+    # Conversion of the image
+    image = np.array(image_thumb)
```

Comment:
I don't get this part. You should not work on a scaled version of the image, because on line 32 you just overwrite _image_ in line 20 that is never used